doc/sphinx/make-your-own-compositor.md

```diff
@@ -0,0 +1,258 @@
+```
+
+### Running
+To run, you can run nested in an X or Wayland session, or from a virtual
+terminal, just like the demo applications in [Explore What Mir Is Capable of
+Using Mir Demos](explore-mir-using-demos.md). For example, if we were to run
+inside an existing Wayland session, we'd run the following command:
+```sh
+WAYLAND_DISPLAY=wayland-99 ./build/demo-mir-compositor 
+```
+You'll see a window housing the compositor with a black void filling it. To
+fill this void with some content, you can run the following from another
+terminal:
+```sh
+WAYLAND_DISPLAY=wayland-99 bomber
+```
+You're free to replace `bomber` with any other Wayland compatible application.
+
+## Startup Applications
+When starting a compositor or display server, you usually want to start a few
+other applications. For example you might want to start a background viewer, a
+terminal, and a status bar. Doing so manually like what we've do would be
+immensely annoying, so let's add an option for users to pass a list of startup
+apps!
+
+It's up to you how you want your command line interface will be. Let's say
+we'll take a list of colon separated program names, then we'll launch each of
+those individually. 
+
+We'll start by creating an instance of `miral::ExternalClientLauncher`, this
+will provide a way to launch programs by name. Then, we'll create a function
+that will take the colon separated string and launch each application
+separately. Finally, we just need to pass the external client launcher and our
+configuration option to {func}`run_with` so that they're registered with the
+compositor.
+
+```diff
+ #include <miral/runner.h>
++#include <miral/configuration_option.h>
++#include <miral/external_client.h>
+ #include <miral/minimal_window_manager.h>
+ #include <miral/set_window_management_policy.h>
+ 
+@@ -9,8 +10,22 @@ int main(int argc, char const* argv[])
+ {
+     MirRunner runner{argc, argv};
+ 
++    miral::ExternalClientLauncher external_client_launcher;
++    
++    auto run_startup_apps = [&](std::vector<std::string> const& apps) 
++    {
++        for(auto const& app : apps)
++        {
++            external_client_launcher.launch(std::vector<std::string>{app}});
++        }
++    };
++
+     return runner.run_with(
+         {
+-            set_window_management_policy<MinimalWindowManager>()
++            set_window_management_policy<MinimalWindowManager>(),
++            external_client_launcher,
++            miral::ConfigurationOption{run_startup_apps, "startup-app", "App to run at startup (can be specified multiple times)"},
+         });
+ }
+```
+
+Now, to start `kgx` and `bomber` on startup, we can do:
+```sh
+./build/demo-mir-compositor --startup-app kgx --startup-app bomber
+```
+
+## Keyboard shortcuts
+What if you accidentally close the only open terminal? Or you want to log out?
+Keyboard shortcuts are a good option that give you great flexibility without
+much coding effort. Let's see how they're implemented in Mir.
```

Comment:
The last section described what was going to happen. This one should too.

It should say that we are going to make `Ctrl-Alt-T` launch a terminal and `Ctrl-AltBkSp` exit.